### PR TITLE
[AddonSettings] Fix setting persistency when acessing raw Settings interface

### LINKED ISSUE #23551: xbmc.Settings.setXxx methods not persisting values (plus, documentation errors)
### Describe the bug
Here is a clear and concise description of what the problem is:

This started as a [help request in the kodi addons forum](https://forum.kodi.tv/showthread.php?tid=373864), but it seems I have found a bug: the `xbmcaddon.Settings.setBool(id, value)` method seems to not persist the value between restarts.

Creating a really simple addon with the following settings:

```xml
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<settings>
	<category label="32000">
      <setting id="firstRun" type="bool" label="32000" default="true" visible="false" />
	</category>
</settings>
```

And these lines of code

```python
import xbmc
import xbmcaddon

addon = xbmcaddon.Addon()
settings = addon.getSettings()
key = 'firstRun'
first_run = settings.getBool(key)
xbmc.log(f'[Test settings] first run: {first_run}')
settings.setBool(key, False)
outcome = "success" if  not settings.getBool(key) else "failure"
xbmc.log(f'[Test settings] {outcome}')
```

I can see that the "firstRun" setting is correctly set to False, but if I restart Kodi it still reads "True" at the first read.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

At successive runs, the logs should read "[Test settings] first run: False"

## Actual Behavior

It still reads "[Test settings] first run: True"

## Possible Fix

switching back to `Addon.setSettingsBool` solves the issue, even if its deprecated in favor of `Settings.setBool`.

### To Reproduce

see above

### Debuglog

The debuglog can be found here: https://paste.kodi.tv/cosepejopo.kodi

### Screenshots
Here are some links or screenshots to help explain the problem:

N/A

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

- [`xbmc/interfaces/legacy/Settings.h`](https://github.com/xbmc/xbmc/blob/master/xbmc/interfaces/legacy/Settings.h#L273-L291) declares `setBool` as void, but the documentation states that it returns a boolean
- [`xbmc/interfaces/legacy/Settings.cpp`](https://github.com/xbmc/xbmc/blob/0d83a0d21c1b1485d969c1b16173890c7595a0a0/xbmc/interfaces/legacy/Settings.cpp#L175-L180) implements `setBool` returning void (so the aforementioned documentation is wrong), and calls `settings->Save()`
- `settings` is of type `CSettingsBase`, an abstract class.

Here I get lost - I'm not so smart to understand C++ :sweat_smile: 

### Your Environment

 Used Operating system:
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

Operating system version/name: Fedora 38
Kodi version: 20.2.0 nexus

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by enen92**
## Description

As reported by https://github.com/xbmc/xbmc/issues/23551 currently when settings are acessed via the new Nexus `Settings` interface, persistency does not occur:

```
import xbmc
import xbmcaddon

addon = xbmcaddon.Addon()
settings = addon.getSettings()
key = 'firstRun'
first_run = settings.getBool(key)
xbmc.log(f'[Test settings] first run: {first_run}')
settings.setBool(key, False)
outcome = "success" if  not settings.getBool(key) else "failure"
xbmc.log(f'[Test settings] {outcome}')
```

Root cause is the fact the `Save()` override pretty much doing nothing :)
The implementation stores the addon pointer (can no longer be const) and saves the settings.

I am a bit unsure if we should just nuke:
```
m_addonId(addon->ID()),
m_addonPath(addon->Path()),
m_addonProfile(addon->Profile()),
```
In the code there's the comment ` // store these values so that we don't always have to access the weak pointer` but as far as I can see we don't access it often (only for a few log lines) and the performance gain should be pretty much negligible. Remove commit: https://github.com/xbmc/xbmc/pull/23558/commits/e02c6f09b904d90749b3a7bf7453b5d456f99616

## Motivation and context
Fix https://github.com/xbmc/xbmc/issues/23551 

## How has this been tested?
As described in the bug report

## What is the effect on users?
Should fix settings not being persisted for addons if acessed via the raw `Settings` class

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

File: xbmc/addons/settings/AddonSettings.cpp [hunk 1]
```diff
@@ -433,6 +433,15 @@ bool CAddonSettings::HasSettings() const
   return IsInitialized() && GetSettingsManager()->HasSettings();
 }
 
+bool CAddonSettings::Save()
+{
+  if (m_addon)
+  {
+    m_addon->SaveSettings();
+  }
+  return true;
```
Comment (lrusak) [inline xbmc/addons/settings/AddonSettings.cpp:442]: This function will always return true, should it return false if m_addon is nullptr?

Comment (enen92) [reply xbmc/addons/settings/AddonSettings.cpp:442]: I think I'd rather change CAddon::SaveSettings() to return the actual success of the operation (there are several steps there that return a status code that is simply being ignored nowadays). If m_addon is nulll something must be wrong and the issue is elsewhere. I'd probably add an assert there and let it crash...

Comment (enen92) [reply xbmc/addons/settings/AddonSettings.cpp]: done in https://github.com/xbmc/xbmc/pull/23558/commits/88a4c418ab36abe37d466c22c995d7249c714837

**Review by Montellese — APPROVED** [review]

**Comment by enen92** [comment]
Jenkins build this please

**Comment by enen92** [comment]
Jenkins build this please

**Comment by MoojMidge** [comment]
Is there any possibility of this and https://github.com/xbmc/xbmc/pull/23864 being backported to the Nexus branch, if there will be a future Kodi v20.x point release?

**Pull request merged**